x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_view_logic.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Status } from '../../../../../common/types/api';
+
+import {
+  FetchEngineApiLogic,
+  FetchEngineApiLogicActions,
+} from '../../api/engines/fetch_engine_api_logic';
+
+import { EngineNameLogic } from './engine_name_logic';
+
+export interface EngineViewActions {
+  fetchEngine: FetchEngineApiLogicActions['makeRequest'];
+}
+
+export interface EngineViewValues {
+  engineData: typeof FetchEngineApiLogic.values.data;
+  engineName: typeof EngineNameLogic.values.engineName;
+  fetchEngineApiStatus: typeof FetchEngineApiLogic.values.apiStatus;
```

Comment:
suggestion: You're keying off apiStatus in a lot of places here, but that value isn't really meant for public consumption--it's essentially an internal state for the API logic. The logic exposes `error`, `status` and `data` as the key primitives we should be working with in most cases, not the encapsulating `apiStatus`. I think you can simplify a number of checks and data pass-alongs throughout this PR if you keep with those primitives instead, but maybe I'm missing something.

Reply:
I had this using `status` initially, but switched to the `apiStatus` to use a single value for the status and the error so that I could render the 404. But I can just connect both the `status` and `error` if that is more idiomatic.

Reply:
That would be better. The reason we have those extra ones exposed is both memoization, and to allow us to futz around with `apiStatus` internally to change things without breaking the external behavior (we had to do that a couple times when initially implementing this).